### PR TITLE
Add tests for WebApplicationFactory<T> issues with minimal hosting

### PR DESCRIPTION
**Pull request by martincostello**
**PR Title**

Add tests for `WebApplicationFactory<T>` issues.

**PR Description**

Add unit tests that replicate #33876 and #33889.

### CODE REVIEW

File: src/Mvc/test/Mvc.FunctionalTests/SimpleWithWebApplicationBuilderTests.cs [hunk 1]
```diff
@@ -1,31 +1,96 @@
...
+
+            // Act
+            var content = await client.GetStringAsync("http://localhost/");
+
+            // Assert
+            Assert.Equal(expected, content);
+        }
+
+        [Fact]
+        [QuarantinedTest("https://github.com/dotnet/aspnetcore/issues/33889")]
```
Comment (martincostello) [inline src/Mvc/test/Mvc.FunctionalTests/SimpleWithWebApplicationBuilderTests.cs:38]: I wasn't sure if this was the right syntax or not for a test known to be failing due to an active issue so it gets ignored and doesn't break the build.

**Comment by javiercn** [comment]
@martincostello we typically don't add these types of tests.

I'll let @davidfowl handle this, since he's been active in the area.

**Comment by davidfowl** [comment]
I think I understand these issues now.

**Comment by davidfowl** [comment]
Superseded by https://github.com/dotnet/aspnetcore/pull/34794

**Pull request closed without merging**